src/ocspdash/web/blueprints/api.py

```diff
@@ -25,25 +26,44 @@
 api = Blueprint('api', __name__)
 
 
-@api.route('/register', methods=['POST'])
+@api.route('/register', methods=['POST'])  # noqa: C901
 def register_location_key():
     """Register a public key for an invited location."""
     # TODO: error handling (what if no invite, what if duplicate name, etc.)
-    unverified_claims = jwt.get_unverified_claims(request.data)
-    unverified_public_key = b64decode(unverified_claims['pk']).decode('utf-8')
+    print(request.data)
+    try:
+        unverified_claims = jwt.get_unverified_claims(request.data)
+    except jwt.JWTError:
+        abort(HTTPStatus.BAD_REQUEST, 'malformed JWT')
+
+    try:
+        unverified_public_key = b64decode(unverified_claims['pk']).decode('utf-8')
+    except KeyError:
+        abort(HTTPStatus.BAD_REQUEST, "'pk' missing from claims")
+    except (binascii.Error, UnicodeError):
+        abort(HTTPStatus.BAD_REQUEST, "failed to decode 'pk' claim")
 
     try:
         claims = jwt.decode(request.data, unverified_public_key)
     except JWTError:
-        return abort(400)  # bad input
+        abort(HTTPStatus.BAD_REQUEST, 'failed to validate JWT')
 
-    public_key = claims['pk']
-    invite_token = b64decode(claims['token'])
+    try:
+        public_key = claims['pk']
+    except KeyError:
+        abort(HTTPStatus.BAD_REQUEST, "'pk' misisng from claims")
 
-    new_location = manager.process_location(invite_token, public_key)
+    try:
+        invite_token = b64decode(claims['token'])
+    except KeyError:
+        abort(HTTPStatus.BAD_REQUEST, "'token' missing from claims")
+    except binascii.Error:
```

Comment:
that's an awful name for an error

Reply:
I take what the stdlib provides.

Reply:
namespaces, those are a honking good thing